File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int maxWidth = 0;
        int maxHeight = 0;
        
        // 모든 명함을 탐색하면서 가로, 세로를 정렬
        for (int[] size : sizes) {
            int w = Math.max(size[0], size[1]);  // 큰 값을 가로로
            int h = Math.min(size[0], size[1]);  // 작은 값을 세로로
            
            // 가로와 세로의 최대값을 갱신
            maxWidth = Math.max(maxWidth, w);
            maxHeight = Math.max(maxHeight, h);
        }
        
        // 지갑 크기는 가장 큰 가로와 세로의 곱
        return maxWidth * maxHeight;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_w, max_h = 0, 0

    for w, h in sizes:
        w, h = max(w, h), min(w, h)  # 큰 쪽을 가로, 작은 쪽을 세로
        max_w = max(max_w, w)
        max_h = max(max_h, h)

    return max_w * max_h

